js/privatebin.js

```diff
@@ -721,32 +758,53 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
         }
 
         /**
-         * check if cipher data was supplied
+         * returns the paste data (inlduing the cipher data)
          *
-         * @name   Model.getCipherData
-         * @function
-         * @return boolean
-         */
-        me.hasCipherData = function()
-        {
-            return (me.getCipherData().length > 0);
-        }
-
-        /**
-         * returns the cipher data
-         *
-         * @name   Model.getCipherData
+         * @name   Model.getPasteData
          * @function
+         * @param {function} callback (optional) Called when data is available
+         * @param {function} useCache (optional) Whether to use the cache or
+         *                            force a data reload. Default: true
          * @return string
          */
-        me.getCipherData = function()
+        me.getPasteData = function(callback, useCache)
         {
-            return $cipherData.text();
+            // use cache if possible/allowed
+            if (useCache !== false && pasteData !== null) {
+                //execute callback
+                if (typeof callback === 'function') {
+                    return callback(pasteData);
+                }
+
+                // alternatively just using inline
+                return pasteData;
+            }
+
+            // reload data
+            Uploader.prepare();
```

Comment:
It's done in other parts of the code too. Works well, but yes, could be renamed. It's just a wrapper around jQueries ajax requests. (Strictly speaking every ajax request is an upload, of course, but well…)
However, I think we can do this once this is merged, as it does not only affect this code part.